src/ToricVarieties/AlgebraicCycles/constructors.jl

```diff
@@ -0,0 +1,196 @@
+
+# Examples
+```jldoctest
+julia> P2 = projective_space(NormalToricVariety, 2)
+A normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> d = ToricDivisor(P2, [1,2,3])
+A torus-invariant, non-prime divisor on a normal toric variety
+
+julia> AlgebraicCycle(d)
+An algebraic cycle on a normal toric variety given by 6*V(x3)
+```
+"""
+function AlgebraicCycle(d::ToricDivisor)
+    v = toric_variety(d)
+    if istrivial(d)
+        return AlgebraicCycle(v,zero(chow_ring(v)))
+    end
+    coeffs = coefficients(d)
+    indets = gens(chow_ring(v))
+    return AlgebraicCycle(v, sum(coeffs[k]*indets[k] for k in 1:length(coeffs)))
+end
+
+
+@doc Markdown.doc"""
+    AlgebraicCycle(c::ToricDivisorClass)
+
+Construct the algebraic cycle corresponding to the toric divisor class `c`.
+
+# Examples
+```jldoctest
+julia> P2 = projective_space(NormalToricVariety, 2)
+A normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> tdc = ToricDivisorClass(P2, [2])
+A divisor class on a normal toric variety
+
+julia> AlgebraicCycle(tdc)
+An algebraic cycle on a normal toric variety given by 2*V(x3)
+```
+"""
+AlgebraicCycle(c::ToricDivisorClass) = AlgebraicCycle(toric_divisor(c))
+
+
+@doc Markdown.doc"""
+    AlgebraicCycle(l::ToricLineBundle)
+
+Construct the toric algebraic cycle corresponding to the toric line bundle `l`.
+
+# Examples
+```jldoctest
+julia> P2 = projective_space(NormalToricVariety, 2)
+A normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> l = ToricLineBundle(P2, [2])
+A toric line bundle on a normal toric variety
+
+julia> polynomial(AlgebraicCycle(l))
+2*x3
+```
+"""
+AlgebraicCycle(l::ToricLineBundle) = AlgebraicCycle(toric_divisor(l))
+
+
+@doc Markdown.doc"""
+    AlgebraicCycle(cc::CohomologyClass)
+
+Construct the toric algebraic cycle corresponding to the cohomology class `cc`.
+
+# Examples
+```jldoctest
+julia> P2 = projective_space(NormalToricVariety, 2)
+A normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> (x1,x2,x3)=gens(cohomology_ring(P2))
+3-element Vector{MPolyQuoElem{MPolyElem_dec{fmpq, fmpq_mpoly}}}:
+ x1
+ x2
+ x3
+
+julia> cc = CohomologyClass(P2,x1+x2)
+A cohomology class on a normal toric variety given by x1 + x2
+
+julia> AlgebraicCycle(cc)
+An algebraic cycle on a normal toric variety given by 2*V(x3)
+```
+"""
+AlgebraicCycle(cc::CohomologyClass) = AlgebraicCycle(toric_variety(cc), polynomial(cc))
+
+
+#################################
+# 3: Addition, subtraction and scalar multiplication
+#################################
+
+function Base.:+(ac1::AlgebraicCycle, ac2::AlgebraicCycle)
+    if toric_variety(ac1) !== toric_variety(ac2)
+        throw(ArgumentError("The algebraic cyclees must be defined on identically the same toric variety."))
```

Comment:
```suggestion
        throw(ArgumentError("The algebraic cycles must be defined on identically the same toric variety"))
```
Typos, and: I think we put o trailing periods into error messages... Not sure if the Julia or Oscar style guides discuss this (@lkastner ?)

Reply:
Thank you.